lib/ethernet/server.ex

```diff
@@ -49,25 +49,28 @@ defmodule Nerves.IO.Ethernet.Server do
   # try renewing dhcp lease upon expiration unless we've been configured
   # as a static ip in the meantime
   def handle_info(:dhcp_lease_expired, state) do
-    case state.status do
-      "static" -> state
-      _ -> configure_with_dynamic_ip(state)
-    end
-    |> respond :noreply
+    state.status
+    |> renew_dhcp(state)
+    |> respond(:noreply)
   end
 
+  defp renew_dhcp("static", state), do: state
+  defp renew_dhcp(_, state), do: configure_with_dynamic_ip(state)
+
   # called periodically to try to see if a dhcp server came back
   # online
   def handle_info(:ip4ll_dhcp_retry, state) do
     params = raw_dhcp_request(state)
-    case params[:status] do
-      "bound" -> configure_dhcp(state, params)
-      "renew" -> configure_dhcp(state, params)
-      _ -> schedule_ip4ll_dhcp_retry(state)
-    end
-    |> respond :noreply
+
+    params[:status]
+    |> conf_dhcp_on_status(state, params)
+    |> respond(:noreply)
   end
```

Comment:
@ghitchens In this case, IMHO, is more important to show your goals, something like: *Ok, According to the status, configure the DHCP and then response `:noreply`*

I lose the focus of your real intentions seeing the `case` statement between your *data transformation*.